services/provider/server/server.go

```diff
@@ -278,23 +278,32 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
 	}
 
-	// Service for monitoring endpoints
-	scMonitoring := &v1.Service{}
-	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	// Get mgr pod hostIP
+	podList := &corev1.PodList{}
+	err = s.client.List(ctx, podList, client.MatchingLabels(map[string]string{"app": "rook-ceph-mgr"}))
+	if err != nil || len(podList.Items) == 0 {
+		return nil, fmt.Errorf("failed to list pod with rook-ceph-mgr label. %v", err)
+	}
+
+	mgrPod := &podList.Items[0]
+	var port int32
+	for i := range mgrPod.Spec.Containers {
+		if mgrPod.Spec.Containers[i].Name == "mgr" {
+			if mgrPod.Spec.Containers[0].Ports[i].Name == "http-metrics" {
+				port = mgrPod.Spec.Containers[0].Ports[i].ContainerPort
+			}
+		}
 	}
```

Comment:
This is wrong, you cannot use the same index for both the container and the port
```suggestion
	for i := range mgrPod.Spec.Containers {
		if mgrPod.Spec.Containers[i].Name == "mgr" {
			container := &mgrPod.Spec.Containers[i]
			for j := range container.Ports {
				if container.Ports[j].Name == "http-metrics" {
					port = container.Ports[j].ContainerPort
				}
			}
		}
	}
```